### PR TITLE
migrate to Mantine V7

### PR DESCRIPTION
**Pull request by Lxkas**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,15 +11,15 @@
 	"dependencies": {
 		"@emotion/react": "^11.11.1",
```
Comment (Lxkas) [inline package.json:12]: prolly can remove this later

File: src/components/page_components/study_plans/StudyPlanTable.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ interface StudyPlanTableProps {
 	elements: StudyPlanElement[];
 }
 
+// TODO: Make the env files .env.local and .env.production.local or w/e
```
Comment (chinathaip) [inline src/components/page_components/study_plans/StudyPlanTable.tsx:13]: Agreed with this one, and let's try to have the `API_BASE_LINK` declared globally or smt just so we can simply reuse it over and over

**Review by chinathaip — COMMENTED** [review]
Looks Gucci

Can you also paste the links about migrating to matine V7 (i.e. what's new, and what is no longer supported smt like that) just so we have more context?

Thank you :3

Also.. need second opinions from frontend guys @NayHtetKyaw

**Comment by Lxkas** [comment]
@chinathaip https://mantine.dev/guides/6x-to-7x/

**Comment by chinathaip** [comment]
> @chinathaip https://mantine.dev/guides/6x-to-7x/

Coolio, thanks!

**Comment by anascence** [comment]
> Looks Gucci
> 
> Can you also paste the links about migrating to matine V7 (i.e. what's new, and what is no longer supported smt like that) just so we have more context?
> 
> Thank you :3
> 
> Also.. need second opinions from frontend guys @NayHtetKyaw

Great Great!, right now i am reading a lot of mantine documentation to keep up with this project.

**Comment by anascence** [comment]
> @chinathaip https://mantine.dev/guides/6x-to-7x/

thank you krub

**Review by chinathaip — APPROVED** [review]
Seems like we're cool with this, let's merge it..

**Comment by Lxkas** [comment]
> Seems like we're cool with this, let's merge it..

@chinathaip not yet, we took so long, like 4 patches already came out XD

**Comment by chinathaip** [comment]
@Lxkas it broke btw.. can u have a look?

**Pull request merged**